### DISCUSSION TITLE
How to decrease the size of binary?

### DISCUSSION THREAD

**Discussion by lz1998** [tokio-rs/tracing#1925] [Q&A] [answered]
### Version

```toml
[dependencies]
tracing = "0.1"
tracing-subscriber = { version = "0.3", features = [
    "fmt",
    "env-filter",
    "local-time",
] }
```

### Platform

MacOS

#### Without tracing

main.rs
```rust
fn main() {
    println!("Hello, world!");
}
```

```
cargo build --release
cd target/release && ll

-rwxr-xr-x   2 lz1998  staff   441K Feb 13 16:51 untitled1*
```

#### Tracing Without EnvFilter

main.rs
```rust
fn main() {
    println!("Hello, world!");
    tracing_subscriber::fmt()
        .without_time()
        .init();
    tracing::info!("hello");
}
```

```bash
cargo build --release
cd target/release && ll

-rwxr-xr-x   2 lz1998  staff   719K Feb 13 16:50 untitled1*
```

#### Tracing With EnvFilter

main.rs
```rust
fn main() {
    println!("Hello, world!");
    let env = tracing_subscriber::EnvFilter::from("");
    tracing_subscriber::fmt()
        .with_env_filter(env)
        .without_time()
        .init();
    tracing::info!("hello");
}
```

```bash
cargo build --release
cd target/release && ll

-rwxr-xr-x   2 lz1998  staff   1.8M Feb 13 16:44 untitled1*
```

**Comment by davidbarsky** [comment, marked as answer]
There are several ways to reduce the size of a Rust binary. [johnthagen/min-sized-rust](https://github.com/johnthagen/min-sized-rust) is a good resource to start with.

In the case of tracing with tracing-subscriber and `EnvFilter`, tracing-subscriber is a _large_ crate and `EnvFilter` pulls in the `regex` crate. If you don't need all the features of `EnvFilter`, you can make use of [`Targets`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/targets/struct.Targets.html).

Does that answer your question or are you asking about something else?
